### ISSUE TITLE
How to run the green function example

### ISSUE THREAD

**Issue by pdelamora** [TRIQS/triqs#216]
I am trying to run the 
- Manipulating local Green’s functions

and I do not know what this means;
- from pytriqs.gf.local import GfReFreq, Omega, Wilson, inverse

so I write in the terminal;
- from ../TRIQS/build/pytriqs/gf/local import GfReFreq, Omega, Wilson, inverse

and I get;
- from: can't read /var/mail/../TRIQS/build/pytriqs/gf/local

**Comment by tayral** [comment]
"from pytriqs.gf.local import GfReFreq, Omega, Wilson, inverse" should 
be run into a **python prompt** -- not the terminal: it is a python 
command to import the modules one needs for this tutorial. For more info 
on python modules, see e.g https://docs.python.org/2/tutorial/modules.html

**Comment by pdelamora** [comment]
Thank you Tayral
It partially worked, but now my problem is that I do not know what the things mean, for example;
- pytriqs.gf.local import 

I ran
- python

then I tried;
- >>> import numpy

it worked fine
then;
- >>> from pytriqs.gf.local import GfReFreq, Omega, Wilson, inverse

ImportError: No module named pytriqs.gf.local
- >>> from TRIQS/src/pytriqs/gf/local import GfReFreq, Omega, Wilson, inverse

SyntaxError: invalid syntax
- >>> from TRIQS.src.pytriqs.gf.local import GfReFreq, Omega, Wilson, inverse

ImportError: No module named TRIQS.src.pytriqs.gf.local

**Comment by tayral** [comment]
You should write "pytriqs" instead of "python". Please take a look at 
the documentation for more details: 
http://ipht.cea.fr/triqs/1.2/tour/getting_started/get_started.html.

**Comment by pdelamora** [comment]
Thanks Tayral,
Unfortunately with ipytriqs I get the same;
first I ran
- ../TRIQS/src/pytriqs

and I get
- bash: ../TRIQS/src/pytriqs: Is a directory

so instead I tryed
- ../TRIQS/build/bin/ipytriqs

and with
- import numpy

it is OK
but with
- from pytriqs.gf.local import GfReFreq, Omega, Wilson, inverse

I get an error, the last line says;
- No module named pytriqs.gf.local

**Comment by pdelamora** [comment]
Tayral,
You say that I should look at 
 ."get started" and it takes me to

**Comment by pdelamora** [comment]
Tayral,
You say that I should look at
- "get started" 

and it takes me to 
-  go to our Python tour 

where I see examples  and all start with 
- from <something> import <something else>

For example, in DMFT
- from pytriqs.gf.local import *
- from pytriqs.gf.local import *
- from pytriqs.operators import *
- from pytriqs.archive import *
- import pytriqs.utility.mpi as mpi

and I look for the meaning of this <something> to see what to put there
Is this something of TRIQS or python?
Can someone tell me?

**Comment by pdelamora** [comment]
Tayral,
Sorry, I used; "<" and ">" and the result is not what I wanted, 
so here I write it again in a slightly different way

You said that I should look at
- get started

and this takes me to
- go to our Python tour

where I see examples and all of them start with
- from "aaa" import "bbb"

For example, in DMFT

    from pytriqs.gf.local import *
    from pytriqs.gf.local import *
    from pytriqs.operators import *
    from pytriqs.archive import *
    import pytriqs.utility.mpi as mpi

and I look for the meaning of this "aaa" (for example; pytriqs.gf.local) to see what to put there
Is this "aaa" something of TRIQS or python?
Can someone tell me?

**Comment by aichhorn** [comment]
LMGTFY. This is how you import modules in python:
http://www.tutorialspoint.com/python/python_modules.htm

TRIQS is, from that point of view, a collection of python modules that you can load and use in python. The executable pytriqs is nothing else than a python interpreter with a defined PYTHONPATH in order to make it easier to load the modules. And yes, everything that starts with pytriqs in the module path is from TRIQS. Other things, like numpy, are not.

**Comment by hbaromega** [comment]
I think, you're probably not getting the issue. I am also encountering the same problem. 

I was trying to execute http://ipht.cea.fr/triqs/1.x/tutorials/gfs/impinbath.py . 

I get a fix when I use `sys.path.append('/home/hbar/Documents/CODES_DMFT/TRIQS/src/')`

`sys.path` result in my system includes

```
'/home/hbar/Documents/CODES_DMFT/TRIQS',
'/home/hbar/Documents/CODES_DMFT/TRIQS/src/'
```

However, I get another similar error that says, 

`ImportError: No module named gf` .

I think, I am doing a mistake in defining the path. But google or your link doesn't help me much !

**Comment by tayral** [comment]
Hi hbaromega,
What command are you using here? "python" or "pytriqs"? If TRIQS is installed correctly, and if you are using "pytriqs", this error should not occur.

**Comment by hbaromega** [comment]
I had been using 'pytriqs' or 'ipytriqs'. 

Now I did a re-installation. The error is gone. But no output as well !! 

Btw I am connected to a remote computer.

**Comment by aichhorn** [comment]
Do you mean with 'no output' that you see no figure on your screen? The problem could be that python/pytriqs closes all windows when it exists. So if you want to keep the figures open, then you have to run the script in a way that pytriqs does not exit:
pytriqs -i script.py
Then, after all commands in the script were executed, you keep a python shell open, and with it all windows hopefully.
The behavior of matplotlib also depends a little bit on the graphics frontend that you are using. Do you get an error message related to the frontend, or nothing?

**Comment by hbaromega** [comment]
I still don't see any graphics. I do not see error message using `pytriqs`, but with `ipytriqs` I get the following warning message related to frontend : 

```
/usr/local/lib/python2.7/dist-packages/IPython/frontend.py:30: UserWarning: The top-level `frontend` package has been deprecated. All its subpackages have been moved to the top `IPython` level.
  warn("The top-level `frontend` package has been deprecated. "
Python 2.7.6 (default, Mar 22 2014, 22:59:56) 
```

Could it be possible that I'm using a wrong version of python?

**Comment by pseth** [comment]
Change the last lines in the example to this:
```
from pytriqs.plot.mpl_interface import *
oplot( g['d','d'], '-o', RI = 'S', x_window  = (-1.8,1.8), name = "Impurity" )
oplot( g['s','s'], '-x', RI = 'S', x_window  = (-1.8,1.8), name = "Bath" )
plt.show()
```
The example is written for ipython notebooks where the image will show up automatically. From a terminal, you need to force the show().

**Comment by hbaromega** [comment]
@pseth ,thanks a lot. It works. :) 
Now can I use this in `ipython notebook` or maybe in `ipytriqs_notebook` ?

**Comment by pseth** [comment]
You should always use the pytriqs calls to ensure that the modules are all in the path, so `ipytriqs_notebook`.

To see things in the notebook, you will need to run `%matplotlib inline`.

**Comment by hbaromega** [comment]
Thanks a lot. So for my example file, I got the desired result after doing the following (I hope that's the appropriate way).

In terminal : 
`ipytriqs_notebook & ` 

Then in the browser (or other viewer) I clicked `New` and selected `Python2`.  In the new Notebook, I executed  these two:
`%matplotlib inline`
`%load 'impinbath.py'`

If there's no syntax or other related stuff, I think, we can announce the issue to be 'solved'. Thanks. :+1:
